File: examples/ta12200.cxx
/*
 * Author: Jon Trulson <jtrulson@ics.com>
 * Copyright (c) 2014 Intel Corporation.
 *
 * Permission is hereby granted, free of charge, to any person obtaining
 * a copy of this software and associated documentation files (the
 * "Software"), to deal in the Software without restriction, including
 * without limitation the rights to use, copy, modify, merge, publish,
 * distribute, sublicense, and/or sell copies of the Software, and to
 * permit persons to whom the Software is furnished to do so, subject to
 * the following conditions:
 *
 * The above copyright notice and this permission notice shall be
 * included in all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
 * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
 * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
 * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
 * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
 * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

#include <unistd.h>
#include <iostream>
#include <signal.h>
#include "ta12200.h"

using namespace std;

int shouldRun = true;

void sig_handler(int signo)
{
  if (signo == SIGINT)
    shouldRun = false;
}


int main()
{
  signal(SIGINT, sig_handler);

//! [Interesting]
  // Instantiate a TA12-200 sensor on analog pin A0
  upm::TA12200* electricity = new upm::TA12200(0);
  
  while (shouldRun)
    {
      unsigned int maxVal = electricity->highestValue();
      float current = electricity->milliAmps(maxVal);

      cout << "Max ADC Value: " << maxVal << ", current: " << current
           << "mA" << endl;
      usleep(100000);
    }

//! [Interesting]

  cout << "Exiting..." << endl;

  delete electricity;
  return 0;
}

File: examples/javascript/ta12200.js
/*
* Author: Zion Orent <zorent@ics.com>
* Copyright (c) 2014 Intel Corporation.
*
* This program and the accompanying materials are made available under the
* terms of the The MIT License which is available at
* https://opensource.org/licenses/MIT.
*
* SPDX-License-Identifier: MIT
*/

var electricitySensor = require('jsupm_ta12200');
// Instantiate a TA12-200 sensor on analog pin A0
var myElectricitySensor = new electricitySensor.TA12200(0);

var maxVal, current;
var myInterval = setInterval(function()
{
	maxVal = myElectricitySensor.highestValue();
	current = myElectricitySensor.milliAmps(maxVal);

	outputStr = "Max ADC Value: " + maxVal +
	", current: " + current + "mA";
	console.log(outputStr);
}, 100);

// Print message when exiting
process.on('SIGINT', function()
{
	clearInterval(myInterval);
	myElectricitySensor = null;
	electricitySensor.cleanUp();
	electricitySensor = null;
	console.log("Exiting...");
	process.exit(0);
});
